_inc/client/components/module-settings/module-settings-form.jsx

```diff
@@ -48,25 +48,43 @@ export function ModuleSettingsForm( InnerComponent ) {
 		},
 		onSubmit( event ) {
 			event.preventDefault();
-			this.props.updateOptions( this.state.options );
+			this.props.updateOptions( this.state.options )
+				.then( () => {
+					this.setState( { options: {} } )
+				} );
 		},
 		getOptionValue( optionName ) {
 			const currentValue = this.props.getOptionCurrentValue( this.props.module.module, optionName );
 			return typeof this.state.options[ optionName ] !== 'undefined'
 				? this.state.options[ optionName ]
 				: currentValue;
 		},
+		shouldSaveButtonBeDisabled() {
+			let shouldItBeEnabled = false;
+			// Check if the form is not currently dirty
+			shouldItBeEnabled = ! this.isSavingAnyOption() && this.isDirty();
+			return ! shouldItBeEnabled;
+		},
 		isDirty() {
 			return !! Object.keys( this.state.options ).length;
 		},
+		isSavingAnyOption() {
+			let isSaving = false;
+			// Check if any of the updated options is still saving
+			Object.keys( this.state.options ).forEach( option_name => {
```

Comment:
@oskosk maybe this can be replaced with
```js
isSaving = Object.keys( this.state.options ).some( option_name => this.props.isUpdating( option_name ) );
```
It has better performance since it will return as soon as the condition is true and won't go through the other options. I've tested this in this branch and seems to work fine.